extensibility/ember-storage-json/src/index.ts

```diff
@@ -0,0 +1,44 @@
+import Service from 'ember-polaris-service';
+import type Storage from 'ember-storage';
+
+export type Serialized =
+  | string
+  | number
+  | boolean
+  | null
+  | SerializedArray
+  | SerializedObject;
+
+export interface SerializedObject {
+  [key: string]: Serialized;
+}
+
+export interface SerializedArray extends Array<Serialized> {}
+
+export type Serializer<T> = (value: T) => SerializedObject;
+export type Deserializer<T> = (serialized: SerializedObject) => T;
+
+export default abstract class JSONStorage extends Service {
+  abstract storage: Storage;
```

Comment:
This requires the app to inject (and therefore compose over) the store, rather than extending it directly